### ISSUE TITLE
Views attached to ApplicationRef cannot be detached from Change Detection unless they're dettached from ApplicationRef itself

### ISSUE THREAD

**Issue by edusperoni** [angular/angular#46690]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

If you create a view and add it to `ApplicationRef` directly, then calling `changeDetecorRef.detach()` will have no effect. It's expected that when detaching the view from change detection, CD does not run, even if it's directly attached to ApplicationRef.

I believe this is related to the fact that `ApplicationRef` does a manual `detectChanges()` call without checking the LView's DIRTY/ATTACHED flags.

https://github.com/angular/angular/blob/3a60063a54d850c50ce962a8a39ce01cfee71398/packages/core/src/application_ref.ts#L986-L988

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-ivy-vvznhk?devToolsHeight=33&file=src/app/app.component.ts

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular 14, but is present in all versions, I believe. See the stackblitz example for details
```

**Comment by pkozlowski-opensource** [comment]
A simpler reproduction scenario using only the public API surface: https://stackblitz.com/edit/angular-ivy-punutw?file=src%2Fapp%2Fapp.component.ts,src%2Fmain.ts

I think that there are several things going on here:
* indeed, `ApplicationRef` doesn't check for flags indicating that a view is detached from the change detection tree;
* dynamic creation of components creates 2 separate views (a root view and a component view).

Needs a bit more investigation but yes, there is something fishy going on here.
